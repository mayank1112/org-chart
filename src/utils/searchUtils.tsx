import React, {ReactElement} from "react";

export const isTeamMatchingKeyword = (teamName: string, keyword: string): boolean => {
    return teamName.toLowerCase().indexOf(keyword.toLowerCase()) !== -1;
};

export const getHighlightedText = (text: string, highlight: string): ReactElement => {
    const parts: string[] = text.split(new RegExp(`(${highlight})`, 'gi'));
    return (
        <span>
            { parts.map((part, iteration) =>
                part
                    ? <span
                        key={iteration}
                        style={
                            part.toLowerCase() === highlight.toLowerCase()
                            ? { fontWeight: 'bold', fontStyle: 'italic', textDecoration: 'underline' }
                            : {}
                        }
                >{part}</span>
                    : null
            )}
        </span>
    );
};
