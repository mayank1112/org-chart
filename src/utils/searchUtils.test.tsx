import {getHighlightedText, isTeamMatchingKeyword} from "./searchUtils";

describe('isTeamMatchingKeyword', () => {
  test('should return true if keyword is present in text', () => {
    const isMatched = isTeamMatchingKeyword('test word', 'tESt Wo');
    expect(isMatched).toBeTruthy();
  });
  test('should return false if keyword is not present in text', () => {
    const isMatched = isTeamMatchingKeyword('test word', 'keyword');
    expect(isMatched).toBeFalsy();
  });
});

describe('getHighlightedText', () => {
  test('should return text if keyword is not matched', () => {
    const outputHTML = getHighlightedText('test word', 'keyword');
    expect(outputHTML).toMatchSnapshot();
  });
  test('should return highlighted text if keyword matches', () => {
    const outputHTML = getHighlightedText('test word', 'tESt Wo');
    expect(outputHTML).toMatchSnapshot();
  });
});
