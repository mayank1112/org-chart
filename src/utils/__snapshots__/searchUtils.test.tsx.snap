// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHighlightedText should return highlighted text if keyword matches 1`] = `
<span>
  <span
    style={
      Object {
        "fontStyle": "italic",
        "fontWeight": "bold",
        "textDecoration": "underline",
      }
    }
  >
    test wo
  </span>
  <span
    style={Object {}}
  >
    rd
  </span>
</span>
`;

exports[`getHighlightedText should return text if keyword is not matched 1`] = `
<span>
  <span
    style={Object {}}
  >
    test word
  </span>
</span>
`;
